cognite/client/_api/sequences.py

```diff
@@ -1009,72 +993,108 @@ def delete_range(self, start: int, end: int | None, id: int | None = None, exter
         sequence = self._cognite_client.sequences.retrieve(id=id, external_id=external_id)
         assert sequence is not None
         post_obj = Identifier.of_either(id, external_id).as_dict()
-        post_obj.update(self._process_columns(column_external_ids=[sequence.column_external_ids[0]]))
+        post_obj.update(self._wrap_columns(column_external_ids=sequence.column_external_ids))
         post_obj.update({"start": start, "end": end})
-        for data, _ in self._fetch_data(post_obj):
-            if data:
-                self.delete(rows=[r["rowNumber"] for r in data], external_id=external_id, id=id)
+        for resp in self._fetch_data(post_obj):
+            if rows := resp["rows"]:
+                self.delete(rows=[r["rowNumber"] for r in rows], external_id=external_id, id=id)
 
+    @overload
     def retrieve(
         self,
-        start: int,
-        end: int | None,
-        column_external_ids: SequenceType[str] | None = None,
-        external_id: str | SequenceType[str] | None = None,
-        id: int | SequenceType[int] | None = None,
+        id_or_external_id: int | str,
+        start: int = 0,
+        end: int | None = None,
+        columns: SequenceType[str] | None = None,
+        limit: int | None = None,
+    ) -> SequenceRows:
+        ...
+
+    @overload
+    def retrieve(
+        self,
+        id_or_external_id: MutableSequence[int] | MutableSequence[str],
```

Comment:
MutableSequence is too strict. It excludes types like tuples or other Sequences.

You can use smth like SequenceNotStr from
https://github.com/hauntsaninja/useful_types/blob/main/useful_types/__init__.py if you want to avoid the issue of str also being a Sequence[str].

Reply:
Will fix this when #1436 is merged as that has the correct type.

Reply:
Nice!